### ISSUE TITLE
Catch exceptions in child processes

### ISSUE THREAD

**Issue by dongbohu** [greenelab/cimr#65]
PR #64 is using a simple `multiprocessing.Pool(...)` to do parallel data processing. It doesn't catch  exceptions raised by child process(es). According to:
https://stackoverflow.com/questions/19924104/python-multiprocessing-handling-child-errors-in-parent
it can be done by subclassing `multiprocessing.Process`. 

`concurrent.futures` seems to be another option:
https://docs.python.org/3/library/concurrent.futures.html

We will file another PR for it after #64. Before that, the non-parallel mode (`-parallel 0` option) is a good way to do troubleshooting.

**Comment by dongbohu** [comment]
Fixed in PR #66.
